src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -220,6 +210,10 @@ public CompletableFuture<CompositePurchaseOrder> createPurchaseOrder(CompositePu
         .thenCompose(tenantConfiguration -> validateAcqUnitsOnCreate(compPO.getAcqUnitIds(), requestContext))
         .thenAccept(ok -> checkOrderApprovalPermissions(compPO, cachedTenantConfiguration, requestContext))
         .thenCompose(ok -> setPoNumberIfMissing(compPO, requestContext)
+        .thenCompose(p -> prefixService.isPrefixAvailable(compPO.getPoNumberPrefix(),requestContext))
+        .thenAccept(p -> setPoNumberPrefix(compPO))
```

Comment:
Using `thenAccept` will the lambda be executed before performing the next `thenCompose`?

Reply:
yes, it is working as expected .